src/main/scala/components/memory-async.scala

```diff
@@ -0,0 +1,378 @@
+// Asynchronous memory module
+
+package dinocpu
+
+import chisel3._
+import chisel3.util._
+
+import chisel3.experimental.ChiselEnum
+
+import chisel3.util.experimental.loadMemoryFromFile
+import firrtl.annotations.MemoryLoadFileType
+
+// A Bundle used for representing a memory access by instruction memory or data memory.
+class Request extends Bundle {  
+  val address   = UInt(32.W)
+  val writedata = UInt(32.W)
+
+  // Encodes the operation requested by this access
```

Comment:
Instead of a 3-state variable, let's split this out into a `valid` bit and a binary signal for write/read.

Reply:
Got it, the `valid` bit is added to the request with the `ValidIO`/`DecoupledIO` interfaces when it is needed.